hummingbot/core/utils/trading_pair_fetcher.py

```diff
@@ -48,9 +46,13 @@ async def fetch_all(self):
             module = getattr(importlib.import_module(module_path), class_name)
             args = {}
             args = conn_setting.add_domain_parameter(args)
-            tasks.append(asyncio.wait_for(asyncio.shield(module.fetch_trading_pairs(**args)), timeout=3))
-            fetched_connectors.append(conn_setting.name)
+            safe_ensure_future(self.call_fetch_pairs(module.fetch_trading_pairs(**args), conn_setting.name))
 
-        results = await safe_gather(*tasks, return_exceptions=True)
-        self.trading_pairs = dict(zip(fetched_connectors, results))
         self.ready = True
+
+    async def call_fetch_pairs(self, fetch_fn, exchange_name):
+        # In case trading pair fetching returned timeout, using empty list
+        try:
+            self.trading_pairs[exchange_name] = await fetch_fn
+        except (asyncio.TimeoutError, asyncio.CancelledError, requests.exceptions.RequestException):
```

Comment:
We should log error messages here. If you can't log them because this process starts before the log in, you can move back to after the log in. I'd rather have an error log than a bit quicker pairs fetching.

Reply:
Moved after the login prompt in new commit